File: JS/problem1.js
const readline = require('readline');
const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});
rl.question("Enter the value of n: ", function(n){
	const A = new Array(parseInt(n));
	rl.question("Enter the elements of A: ", function(input){
		const elements = input.split(" ");
		for(let i = 0; i < n; i++){
			A[i] = parseInt(elements[i]);
		}
		let newSize = n;
		for(let i = 0; i < n; i++){
			if(A[i] % 2 == 0){
				newSize++;
			}
		}
		const newArray = new Array(newSize);
		let index = 0;
		for (let i = 0; i < n; i++){
			if(A[i] % 2 == 0){
				newArray[index] = 0;
				index++;
			}
			newArray[index] = A[i];
			index++;
		}
		console.log(newSize);
		for(let i = 0; i < newSize; i++){
			process.stdout.write(newArray[i] + " ");
		}
		rl.close();
	});
});

File: C++/problem1.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    cin >> n;
    int* A = new int[n];
    for(int i = 0; i < n; i++){
        cin >> A[i];
    }
    int newSize = n;
    for(int i = 0; i < n; i++){
        if(A[i] % 2 == 0){
            newSize++;
        }
    }
    int* newArray = new int[newSize];
    int index = 0;
    for (int i = 0; i < n; i++){
        if(A[i] % 2 == 0){
            newArray[index] = 0;
            index++;
        }
        newArray[index] = A[i];
        index++;
    }
    cout << newSize << endl;
    for(int i = 0; i < newSize; i++){
        cout << newArray[i] << " ";
    }
    delete[] A;
    delete[] newArray;
    return 0;
}